src/main/java/com/demo/classroom/Service/AuthService.java

```diff
@@ -70,6 +89,49 @@ public ApiResponse<Void> registerUser(RegistrationDTO request) {
         }
         return createApiResponse(false, Constants.REGISTRATION_FAILED.getMessage());
     }
+    
+    @Transactional
+    public ApiResponse<?> loginUser(LoginDTO request) {
+        
+        boolean validUsername = userRepository.existsByUsername(request.getUsername());
+
+        if (!validUsername) {
+            
+            return createApiResponse(false, Constants.INVALID_USERNAME.getMessage());
+        }
+
+        Authentication authentication = authenticationManager.authenticate(
+            new UsernamePasswordAuthenticationToken(request.getUsername(), request.getPassword())
+        );
+
+        var authUser = userService.loadUserByUsername(request.getUsername());
+
+        User user = userRepository.findByUsername(request.getUsername())
+                .orElseThrow(() -> new UsernameNotFoundException("Invalid username"));
+
+        var userId = user.getId();
+        var teacher = teacherRepository.findByUserId(userId);
+        var student = studentRepository.findByUserId(userId);
+
+        boolean isTeacher = authentication.getAuthorities().stream()
+            .anyMatch(role -> role.getAuthority().equals("ROLE_TEACHER"));
+        boolean isStudent = authentication.getAuthorities().stream()
+            .anyMatch(role -> role.getAuthority().equals("ROLE_STUDENT"));
+            
+        if (isTeacher) {
+            Map<String, String> jwtToken = new HashMap<>();
+            jwtToken.put("accessToken", jwtService.generateToken(authUser, teacher.get().getId()));
+            jwtToken.put("refreshToken", jwtService.generateRefreshToken(authUser));
+            return new ApiResponse<>(true, Constants.TEACHER_LOGIN_SUCCESSFULL.getMessage(), jwtToken);
+        } else if (isStudent) {
+            Map<String, String> jwtToken = new HashMap<>();
+            jwtToken.put("accessToken", jwtService.generateToken(authUser, student.get().getId()));
+            jwtToken.put("refreshToken", jwtService.generateRefreshToken(authUser));
+            return new ApiResponse<>(true, Constants.STUDENT_LOGIN_SUCCESSFULL.getMessage(), jwtToken);
+        } 
+        
+        return createApiResponse(false, Constants.LOGIN_FAILED.getMessage());
+    }
```

Comment:
The loginUser method is doing too much. It should be broken down into smaller methods to make it more readable and maintainable.

Reply:
Done